src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1274,42 +1266,49 @@ export default class Blockchain extends Emittery.Typed<
         Buffer /*codeHash*/
       ])[2];
 
-      let keys: Buffer[] = [];
-      return new Promise((resolve, reject) => {
-        storageTrie
-          .createReadStream()
-          .on("data", data => {
-            keys.push(data.key);
-          })
-          .on("end", () => {
-            // #4 - sort and filter keys
-            const sortedKeys = keys.sort((a, b) => Buffer.compare(a, b));
-
-            // find starting point in array of sorted keys
-            const startKeyBuffer = Data.from(startKey).toBuffer();
-            keys = sortedKeys.filter(key => {
-              if (Buffer.compare(startKeyBuffer, key) <= 0) {
-                return key;
-              }
-            });
+      return new Promise<T>((resolve, reject) => {
+        const startKeyBuffer = Data.from(startKey).toBuffer();
+        const compare = (a: Buffer, b: Buffer) => a.compare(b) < 0;
+
+        const keys: Buffer[] = [];
+        const handleData = ({ key }) => {
+          // ignore anything that comes before our starting point
+          if (startKeyBuffer.compare(key) > 0) return;
```

Comment:
If the key comes _before_ our start key we can just ignore it. No need to sort data we know we'll throw away!